app/src/test/java/com/appunite/loudius/di/CheckModulesTest.kt

```diff
@@ -0,0 +1,42 @@
+package com.appunite.loudius.di
+
+import android.content.Context
+import android.content.SharedPreferences
+import com.appunite.loudius.util.MainDispatcherExtension
+import io.mockk.every
+import io.mockk.mockkClass
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.extension.ExtendWith
+import org.koin.dsl.koinApplication
+import org.koin.test.KoinTest
+import org.koin.test.check.checkModules
+import org.koin.test.mock.MockProvider
+
+@ExtendWith(MainDispatcherExtension::class)
+class CheckModulesTest : KoinTest {
+
+    @Test
+    fun verifyKoinApp() {
```

Comment:
**suggestion**: When do you think this test might fail? 

AFAIK this test might be much more useful if it would test logic that is used in `LoudiusApplication`

Reply:
used the appModules reference to the appmodule excluding viewmodel module as it is causing headache, because of tests complaining about not returning stuff for sharedprefs for some keys that the values are retrieved when the ViewModel is initialized.

 I do not think we are loosing much not checking for ViewModels, as mostly ViewModel is last item in the dependency three, which should have "bigger blocks" being injected, so all the most important validation is happening before viewmodels injection, let me know if that is okay with you and if you agree with it or not...